src/app/pages/virtualization/components/all-instances/instance-list/instance-list.component.ts

```diff
@@ -28,30 +44,76 @@ import { VirtualizationInstancesStore } from 'app/pages/virtualization/stores/vi
   changeDetection: ChangeDetectionStrategy.OnPush,
   standalone: true,
   imports: [
+    MatButton,
+    IxIconComponent,
+    MatMenu,
+    MatMenuItem,
+    MatMenuTrigger,
     TranslateModule,
     SearchInput1Component,
     FakeProgressBarComponent,
     InstanceRowComponent,
     MatCheckboxModule,
     EmptyComponent,
+    RequiresRolesDirective,
     TestDirective,
   ],
 })
 
 export class InstanceListComponent {
-  protected readonly requireRoles = [Role.VirtInstanceWrite];
   protected readonly searchQuery = signal<string>('');
   protected readonly showMobileDetails = signal<boolean>(false);
   protected readonly isMobileView = signal(false);
   protected readonly window = inject<Window>(WINDOW);
   protected readonly selection = new SelectionModel<string>(true, []);
+  protected readonly requiredRoles = [Role.VirtInstanceWrite];
+
+  readonly bulkActionStartedMessage = this.translate.instant('Requested action performed for selected Instances');
 
   protected readonly instances = this.store.instances;
   protected readonly isLoading = this.store.isLoading;
   protected readonly selectedInstance = this.store.selectedInstance;
 
-  protected readonly isAllSelected = computed(() => {
-    return this.selection.selected.length === this.instances().length;
+  get checkedInstancesIds(): string[] {
+    return this.selection.selected;
+  }
+
+  get isAllSelected(): boolean {
+    return this.selection.selected.length === this.filteredInstances().length;
+  }
+
+  get hasCheckedInstances(): boolean {
+    return this.checkedInstancesIds.length > 0;
+  }
+
+  protected readonly checkedInstances = computed(() => {
+    return this.checkedInstancesIds.map((id) => this.instances().find((instance) => instance.id === id));
+  });
+
+  protected readonly isBulkStartDisabled = computed(() => {
+    return this.checkedInstances().every(
+      (instance) => [VirtualizationStatus.Running].includes(instance.status),
+    );
+  });
+
+  protected readonly isBulkStopDisabled = computed(() => {
+    return this.checkedInstances().every(
+      (instance) => [VirtualizationStatus.Stopped].includes(instance.status),
+    );
+  });
+
+  protected readonly activeCheckedInstances = computed(() => {
+    return this.instances().filter(
+      (instance) => [VirtualizationStatus.Running].includes(instance.status)
+        && this.selection.isSelected(instance.id),
+    );
+  });
+
+  protected readonly stoppedCheckedInstances = computed(() => {
+    return this.instances().filter(
+      (instance) => [VirtualizationStatus.Stopped].includes(instance.status)
+        && this.selection.isSelected(instance.id),
+    );
```

Comment:
See if you can extract it into it's own component.

Reply:
@undsoft - not sure what do you mean, exactly?